dbt_meshify/storage/yaml_editors.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, Any
+from collections import OrderedDict
+
+from dbt.contracts.graph.nodes import Group
+from dbt.contracts.results import CatalogTable
+from dbt.node_types import AccessType
+
+
+class DbtMeshYmlEditor:
+    """
+    Class to operate on the contents of a dbt project's dbt_project.yml file
+    to add the dbt-core concepts specific to the dbt linker
+    """
+
+    @staticmethod
+    def add_group_to_yml(group: Group, full_yml_dict: Dict[str, Any]):
+        """Add a group to a yml file"""
+        if full_yml_dict is None:
```

Comment:
do you think we can a) tighten up this logic and/or b) abstract this into methods at all? i think some of the boilerplate read-and-set-up-a-dict and filter-and-write-out the yml stuff might be really similar across methods

Reply:
I do! Ideally, this type of low-level list/dict manipulation can be abstracted into some other structure. I'll have to think a little more as to what that ought to be!